.github/workflows/pypi-publish.yml

```diff
@@ -0,0 +1,32 @@
+# This workflows will upload a Python Package using Twine when a release is created
+# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
+
+name: Upload Python Package
+
+on:
```

Comment:
The travis script uploaded new versions on tag events. Can github actions do that?

Reply:
Reading about how actions parses that now:

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags

Reply:
This sounds like what you're looking for? https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags

```
on:
  push:
    # Sequence of patterns matched against refs/heads
    branches:    
      # Push events on main branch
      - main
      # Push events to branches matching refs/heads/mona/octocat
      - 'mona/octocat'
      # Push events to branches matching refs/heads/releases/10
      - 'releases/**'
    # Sequence of patterns matched against refs/tags
    tags:        
      - v1             # Push events to v1 tag
      - v1.*           # Push events to v1.0, v1.1, and v1.9 tags
```

Reply:
They do have an on-release trigger:

```
# Trigger the workflow on pull request activity
on:
  release:
    # Only use the types keyword to narrow down the activity types that will trigger your workflow.
    types: [published, created, edited]
```

Github Releases are based on tags.  Have you been manually creating these releases or does Github auto-create these when you push new tags?

https://github.com/23andMe/Yamale/releases

Alternatively, we can also trigger on tag push/creation:

```
on:
  push:
    # Sequence of patterns matched against refs/heads
    branches:    
      # Push events on main branch
      - main
    # Sequence of patterns matched against refs/tags
    tags:        
      - v1             # Push events to v1 tag
      - v1.*           # Push events to v1.0, v1.1, and v1.9 tags
```

or

```
on:
  push:
    tags:
    - '*'
```

Reply:
I think that last example looks like what we'd want. The releases are created by hand, usually only when something very interesting goes in. We'd still want to push to pypi for every new tag that's created.

Reply:
Updated to run on push for tags